pkg/chaintracker/chainfollower.go

```diff
@@ -71,29 +70,37 @@ func (c *ChainFollower) Run(started, stopped chan bool, stop chan context.Contex
 
 		// Startup: catch up to the current Best Block (tip)
 		var pos ChainPos
-		if state.BestBlockHash == "" {
-			// No BestBlockHash stored, so we must be starting from scratch.
-			// Walk the blockchain from the genesis block.
-			pos = ChainPos{"", 0, DOGECOIN_GENESIS_BLOCK_HASH}
-		} else {
-			// Walk forwards on the blockchain from BestBlockHash until we reach the tip.
-			pos = ChainPos{state.BestBlockHash, state.BestBlockHeight, ""}
-		}
-		pos = c.followChainToTip(pos)
 
-		// Main loop: catch up to the current Best Block (tip) each time it changes.
-		for {
-			// Wait for Core to signal a new Best Block (new block mined)
-			select {
-			case <-stop:
-				stopped <- true
-				return
-			case <-c.ReceiveBestBlock:
-				log.Println("ChainFollower: received new block signal")
-			}
+		// Get Genesis Block
+		blockhash, err := c.l1.GetBlockHash(1)
```

Comment:
Better to move this inside the `if state.BestBlockHash == "" {` below since that's the only time we need to fetch it

Reply:
I added and squashed :)